common/core/desktop/src/kmx/kmx_xstring.cpp

```diff
@@ -173,43 +173,18 @@ PKMX_WCHAR km::kbp::kmx::decxstr(PKMX_WCHAR p, PKMX_WCHAR pStart)
   {
     return p-1;
   }
+
   else if(*(p-1) == UC_SENTINEL) return p-1;
-  else if(p > pStart+1 && *(p-2) == UC_SENTINEL)
-  {
-    switch(*(p-1))
-    {
-      case CODE_ANY:
-      case CODE_NOTANY:
-      case CODE_USE:
-      case CODE_DEADKEY:
-      case CODE_CLEARCONTEXT:
-      case CODE_CALL:
-      case CODE_CONTEXTEX:
-      case CODE_RESETOPT:
-      case CODE_SAVEOPT:
-        return p-2;
-    }
-  }
-  else if(p > pStart+2 && *(p-3) == UC_SENTINEL)
+  else
   {
-    switch(*(p-2))
+    for (int i = 1; i < 5; i++)
     {
-      case CODE_INDEX:
-      case CODE_SETOPT:
-      case CODE_SETSYSTEMSTORE:
-        return p-3;
+      //   p right of pstart &&  *(p - i) == UC_SENTINEL  &&  next CODE_ right of UC_SENTINEL ( looked up in CODE__SIZE+1) has value i
+      if ((p >= pStart + i)  && (*(p - i) == UC_SENTINEL) && (CODE__SIZE[*(p - i + 1)] + 1 == i))
+        return (p - (CODE__SIZE[*(p - i + 1)] + 1));
```

Comment:
This would be easier to humanly digest as:

```suggestion
        return (p - i);
```

Reply:
...changed it